test/CMakeLists.txt

```diff
@@ -1,9 +1,18 @@
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.24)
 
 project(concurrencppTests LANGUAGES CXX)
 
 include(../cmake/coroutineOptions.cmake)
 
+if(NOT DEFINED CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+  get_cmake_property(GENERATOR_IS_MULTI_CONFIG GENERATOR_IS_MULTI_CONFIG)
+  if(GENERATOR_IS_MULTI_CONFIG)
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/$<CONFIG>/bin)
+  else()
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
+  endif()
+endif()
+
```

Comment:
Defining this here is pointless, since `concurrencpp` will never see this value set and the point would've been for that and the tests to be in the same folder. Like I said many times before, just abandon this approach. Append to `PATH` in CI and let the user set the cache variables that they are supposed to solely be able to control from the command line IF they prefer that kind of workflow.

Reply:
You are incorrect, `concurrencpp` does see it since it is fetched as a subproject of the test project. e.g.:
```pwsh
PS D:\dev\concurrencpp\build> dir .\bin\

    Directory: D:\dev\concurrencpp\build\bin

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---          10/21/2022  8:54 AM         342528 async_lock_tests.exe
-a---          10/21/2022  8:54 AM        2177408 async_lock_tests.ilk
-a---          10/21/2022  8:54 AM        3305472 async_lock_tests.pdb
-a---          10/21/2022  8:54 AM         760320 concurrencpp.dll
-a---          10/21/2022  8:54 AM        5153536 concurrencpp.ilk
-a---          10/21/2022  8:54 AM        5058560 concurrencpp.pdb
-a---          10/21/2022  8:54 AM         484352 coroutine_promise_tests.exe
-a---          10/21/2022  8:54 AM        2365792 coroutine_promise_tests.ilk
-a---          10/21/2022  8:54 AM        3641344 coroutine_promise_tests.pdb
...
```

Reply:
Right, this test project is doing the weird thing where it vendors the parent project. Agh! I blame myself from ~2 years ago.